src/test/java/io/cdap/plugin/RunConfigTest.java

```diff
@@ -17,73 +17,104 @@
                                                             Schema.Field.of("input", Schema.of(Schema.Type.STRING)));
+  private static final String MOCK_STAGE = "mockStage";
+  private static final RunConfig VALID_CONFIG = new RunConfig(
+    "java -jar /home/user/Example.jar",
+    "input",
+    "50 true",
+    "output",
+    "string",
+    null
+  );
+
+  @Test
+  public void testValidConfig() {
+    MockFailureCollector failureCollector = new MockFailureCollector(MOCK_STAGE);
+    VALID_CONFIG.validate(failureCollector, inputSchema);
+    Assert.assertTrue(failureCollector.getValidationFailures().isEmpty());
+  }
 
   @Test
   public void testInvalidCommandToExecute() throws Exception {
-    Run.RunConfig config = new Run.RunConfig("java -jar", "input", "50 true", "output", "string", null);
-    MockPipelineConfigurer configurer = new MockPipelineConfigurer(inputSchema);
+    MockFailureCollector failureCollector = new MockFailureCollector(MOCK_STAGE);
+    RunConfig config = RunConfig.builder(VALID_CONFIG)
+      .setCommandToExecute("java -jar")
+      .build();
+    List<List<String>> paramName = Arrays.asList(
+      Collections.singletonList(RunConfig.COMMAND_TO_EXECUTE),
+      Collections.singletonList(RunConfig.COMMAND_TO_EXECUTE));
+
     try {
-      new Run(config).configurePipeline(configurer);
-      Assert.fail();
-    } catch (IllegalArgumentException e) {
-      Assert.assertEquals("Error while accessing the binary. Please make sure that the 'Command to Execute' is in " +
-                            "the expected format. 'java -jar'", e.getMessage());
+      config.validate(failureCollector, inputSchema);
+    } catch (ValidationException e) {
+      assertValidationFailed(failureCollector, paramName);
     }
   }
 
   @Test
   public void testInvalidBinaryType() throws Exception {
-    Run.RunConfig config = new Run.RunConfig("java -jar /home/user/Example.dll", "input", "50 true", "output",
-                                             "string", null);
-    MockPipelineConfigurer configurer = new MockPipelineConfigurer(inputSchema);
+    MockFailureCollector failureCollector = new MockFailureCollector(MOCK_STAGE);
+    RunConfig config = RunConfig.builder(VALID_CONFIG)
+      .setCommandToExecute("java -jar /home/user/Example.dll")
+      .build();
+    List<List<String>> paramName = Collections.singletonList(
+      Collections.singletonList(RunConfig.COMMAND_TO_EXECUTE));
+
     try {
-      new Run(config).configurePipeline(configurer);
-      Assert.fail();
-    } catch (IllegalArgumentException e) {
-      Assert.assertEquals("Binary type 'dll' is not supported. Supported executable types are: 'exe, sh, bat and jar'" +
-                            ".", e.getMessage());
+      config.validate(failureCollector, inputSchema);
+    } catch (ValidationException e) {
+      assertValidationFailed(failureCollector, paramName);
```

Comment:
this test will not fail if validation passes but it should

Reply:
Fixed